Core/Src/main.c

```diff
@@ -64,8 +71,10 @@ void SystemClock_Config(void);
   */
 int main(void)
 {
-  /* USER CODE BEGIN 1 */
 
+  /* USER CODE BEGIN 1 */
+  txData[0] = 0;
+  txData[1] = 0b1000;
```

Comment:
Alright so it looks like in terms of sending and receiving data, we are using an array of size 2... now I think I know why you may have done this but we should take a closer look at the data sheet, especially figure 6-1 below: 
![image](https://user-images.githubusercontent.com/69175242/211224570-0d5d422a-6b78-4b57-ae99-8358822fed79.png)


Based on this image, looking at the `MCU Transmitted Data` and `MCU Received Data` part, it is saying we are transmitting and receiving 3 bytes, not 2. Even though some of the data is 'Don't Care' or unknown (signified with a ?) we still need to receive that data then parse it out later.

Also, it seems like you are missing the 'start bit' within the first byte in `txData` but the configuration bits `0b1000` look good to me.

Also also, in general, it is best practice to display these types of values in the HEX numbering system. Reason being, if we starting sending configuration bits that are 32 bits long then having 32 ones and zeros would be very unreadable.